runtime/lua/vim/lsp/buf.lua

```diff
@@ -472,15 +472,17 @@ function M.list_workspace_folders()
   return workspace_folders
 end
 
---- Add the folder at path to the workspace folders. If {path} is
---- not provided, the user will be prompted for a path using |input()|.
-function M.add_workspace_folder(workspace_folder)
+--- Add the folder at path to the clients or client workspace folders.
+--- If {path} and {client_id} is not provided, the user will be
+--- prompted for a path using |input()|.
+function M.add_workspace_folder(workspace_folder, client_id)
```

Comment:
This reads as if the prompt only happens if _both_ arguments are not provided, but it's not depending on `client_id`

Reply:
right.

Reply:
thinking about `opts` this is a table include the options . if we don't plan to add more options for this function. just one param `client_id` is enough .

Reply:
thinking about `opts` this is a table include the options . if we don't plan to add more options for this function. just one param `client_id` is enough .

Reply:
It would help to know what motivated this PR. I doubt users who call this method manually are going to know the client id. Being able to provide the `name` of the client would make it easier to use. 

But see also https://github.com/neovim/neovim/pull/20779#pullrequestreview-1152265645
We should have a high level picture of how we want to deal with multiple workspaces in general to make sure the pieces fit together

Reply:
The motivation comes from my recent observation of some lsp-related plug-ins such as emacs lsp-bridge /lsp-mode vim yegappan/lsp. They don't use root dir at all, which wastes time. Read the lsp standard. workspaceFolders is the recommended rootpath which has been removed. It is helpful for some lsp plugins to support inserting workspacefolder for specific client here. For example, save the client id after starting a server, and only need to insert this client when opening a new project.

ref [lspconfig](https://github.com/neovim/nvim-lspconfig/commit/b03ff434790b42b03a6c34b97ba0c5dd0afe5142)  
if this function support insert workspaceFolder to a client there will be easy .

Reply:
> They don't use root dir at all, which wastes time.

How does that waste time?

>  Read the lsp standard. 

I did?

> workspaceFolders is the recommended rootpath which has been removed

vim.lsp.start_client sets `workspaceFolders`. The `root_dir` is a concept of the client to simplify the common case of having 1 project folder. 

> It is helpful for some lsp plugins to support inserting workspacefolder for specific client here.

Maybe, but I'm not sure if this here is the best approach to do that.

> save the client id after starting a server, and only need to insert this client when opening a new project.

That works in code, but the method here was intended for manual calls (hence the input prompt)